### PR TITLE
Test: A few tests for module

### PR DESCRIPTION
**Pull request by DongLieu**

### CODE REVIEW

**Comment by catShaark** [comment]
plz provide more description about this pr (what functions is being tested....).

**Comment by DongLieu** [comment]
This PR includes:
- update testing/simapp and keeper test
- test EndBlocker
- test genesis
- test msg
- test AccAddrAndValAddrFromStrings

File: testing/simapp/export.go [hunk 1]
```diff
@@ -147,11 +140,11 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
 	})
 
 	// iterate through unbonding delegations, reset creation height
-	app.StakingKeeper.IterateUnbondingDelegations(ctx, func(_ int64, ubd stakingtypes.UnbondingDelegation) (stop bool) {
-		for i := range ubd.Entries {
-			ubd.Entries[i].CreationHeight = 0
+	app.StakingKeeper.IterateUnbondingDelegations(ctx, func(_ int64, unlock stakingtypes.UnbondingDelegation) (stop bool) {
+		for i := range unlock.Entries {
```
Comment (catShaark) [inline testing/simapp/export.go:144]: why changing udb -> unlock

File: testing/simapp/simd/cmd/cmd_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+	svrcmd "github.com/cosmos/cosmos-sdk/server/cmd"
+	"github.com/cosmos/cosmos-sdk/x/genutil/client/cli"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/ibc-go/v6/testing/simapp"
+	"github.com/cosmos/ibc-go/v6/testing/simapp/simd/cmd"
+)
+
+func TestInitCmd(t *testing.T) {
```
Comment (catShaark) [inline testing/simapp/simd/cmd/cmd_test.go:15]: this test fails

Comment (catShaark) [reply testing/simapp/simd/cmd/cmd_test.go:15]: please remove it

File: testutil/address.go [hunk 3]
```diff
@@ -16,3 +17,8 @@ func GenValAddress() sdk.ValAddress {
 
 	return sdk.ValAddress(priv.PubKey().Address())
 }
+
+func GenPubKey() cryptotypes.PubKey {
```
Comment (catShaark) [inline testutil/address.go:21]: This seems unnecessary

File: testing/simapp/export.go [hunk 4]
```diff
@@ -78,15 +79,11 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
 	// withdraw all delegator rewards
 	dels := app.StakingKeeper.GetAllDelegations(ctx)
 	for _, delegation := range dels {
-		valAddr, err := sdk.ValAddressFromBech32(delegation.ValidatorAddress)
+		delAddr, valAddr, err := types.AccAddrAndValAddrFromStrings(delegation.DelegatorAddress, delegation.ValidatorAddress)
```
Comment (catShaark) [inline testing/simapp/export.go:82]: I think we should only use this ultil func inside multi-staking module package

File: x/multi-staking/types/genesis_test.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/stretchr/testify/require"
+
+	"github.com/realio-tech/multi-staking-module/testutil"
+	"github.com/realio-tech/multi-staking-module/x/multi-staking/types"
+)
+
+func TestGenesisState_Validate(t *testing.T) {
+	multiStakerAddress := testutil.GenAddress()
+	valPubKey := testutil.GenPubKey()
+	valAddr := sdk.ValAddress(valPubKey.Address())
```
Comment (catShaark) [inline x/multi-staking/types/genesis_test.go:16]: ditto

File: testing/simapp/export.go [hunk 6]
```diff
@@ -79,15 +80,16 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
 	// withdraw all delegator rewards
 	dels := app.StakingKeeper.GetAllDelegations(ctx)
 	for _, delegation := range dels {
-		valAddr, err := sdk.ValAddressFromBech32(delegation.ValidatorAddress)
+		delAddr, err := sdk.AccAddressFromBech32(delegation.DelegatorAddress)
```
Comment (catShaark) [inline testing/simapp/export.go:83]: why switching places between these two

File: testing/simapp/export.go [hunk 7]
```diff
@@ -115,11 +117,7 @@ func (app *SimApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs []
 
 	// reinitialize all delegations
 	for _, del := range dels {
-		valAddr, err := sdk.ValAddressFromBech32(del.ValidatorAddress)
-		if err != nil {
-			panic(err)
-		}
-		delAddr, err := sdk.AccAddressFromBech32(del.DelegatorAddress)
+		delAddr, valAddr, err := types.AccAddrAndValAddrFromStrings(del.DelegatorAddress, del.ValidatorAddress)
```
Comment (catShaark) [inline testing/simapp/export.go:120]:
```suggestion
				valAddr, err := sdk.ValAddressFromBech32(del.ValidatorAddress)
		if err != nil {
			panic(err)
		}
		delAddr, err := sdk.AccAddressFromBech32(del.DelegatorAddress)
```

File: x/multi-staking/keeper/abci_test.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+func (suite *KeeperTestSuite) TestEndBlocker() {
+	suite.SetupTest()
+
+	mulStaker := testutil.GenAddress()
+	valAddr := testutil.GenValAddress()
+	const gasDenom = "ario"
```
Comment (neitdung) [inline x/multi-staking/keeper/abci_test.go:19]: this variable could not be const and you can change this variable name to `multiStakingDenomA`

File: x/multi-staking/keeper/abci_test.go [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+			ValAddr:         valAddr.String(),
+		},
+		Entries: []types.UnlockEntry{unlockEntry},
+	}
+
+	suite.msKeeper.SetMultiStakingUnlock(suite.ctx, newUbd)
+	matureUnbondingDelegations := suite.msKeeper.GetMatureUnbondingDelegations(suite.ctx)
+	completionTime := suite.ctx.BlockHeader().Time
+
+	unDT := stakingtypes.UnbondingDelegationEntry{
```
Comment (neitdung) [inline x/multi-staking/keeper/abci_test.go:49]: Please change the variable name for much more information

File: x/multi-staking/keeper/abci_test.go [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+		CreationHeight: suite.ctx.BlockHeight(),
+		CompletionTime: completionTime,
+		InitialBalance: sdk.NewInt(1000),
+		Balance:        sdk.NewInt(1000),
+	}
+
+	unD := stakingtypes.UnbondingDelegation{
```
Comment (neitdung) [inline x/multi-staking/keeper/abci_test.go:56]: Please change the variable name for much more information

File: x/multi-staking/types/genesis_test.go [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+	"github.com/realio-tech/multi-staking-module/testutil"
+	"github.com/realio-tech/multi-staking-module/x/multi-staking/types"
+	"github.com/stretchr/testify/require"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func TestGenesisState_Validate(t *testing.T) {
+	multiStakerAddress := testutil.GenAddress()
```
Comment (neitdung) [inline x/multi-staking/types/genesis_test.go:14]: Please make all variables have the same naming convention

**Comment by catShaark** [comment]
closes by: #120

**Pull request closed without merging**